### PR TITLE
SAK-25439 Better support for right to left (RTL) skins

### PR DESCRIPTION
**Pull request by juanjmerono**

### CODE REVIEW

**Comment by alienresident** [comment]
As far as Morpheus goes this looks good to me. I'll be adding some css/sass commits for this to look cleaner shortly. But I can't merged it.

Thanks,
Mark

**Comment by juanjmerono** [comment]
I can merge it, but it’s my own PR, so lets someone else to take a look and do the merge :)

**Comment by ottenhoff** [comment]
My only question: is it okay to depend on java.awt packages?  Or should we have our own method to decide if the lang is RTL?

        if( "iw".equals(lang) || "ar".equals(lang) || "fa".equals(lang) || "ur".equals(lang) )

**Comment by jonespm** [comment]
Yeah, I don't think we should use the java.awt. I don't see anything thats 100% equivalent. Probably just including our own method somewhere would be best. This says that method isn't even up-to-date.

http://stackoverflow.com/a/20821395/3708872

**Comment by juanjmerono** [comment]
Ok, should we have a sakai property also?, something like portal.rtl.locales=ar,dv,....  (with "ar,dv,fa,ha,he,iw,ji,ps,ur,yi" as default value)

**Comment by buckett** [comment]
Having our own method (possibly pulled from properties) also makes testing much easier as you can set en to be RTL so you can still use the UI. I wouldn't prefix it with portal as it's also something that affects the tools, maybe:
```
locales.rtl=ar,dv,fa,ha,he,iw,ji,ps,ur,yi
```
Having our own method would be nice so that it's easy to change all implementations and see where is using it.

**Comment by juanjmerono** [comment]
I'm wondering also if now with iframeless portal it's still needed add dir tag in other places than BasePortalHandler. This patch adds dir attribute in every html tag renderer, but now only the main portal is going to render this tag, isn't it?

**Comment by juanjmerono** [comment]
I've changed the PR to remove java.awt dependency. Now there is a new property: locales.rtl with default value: ar,dv,fa,ha,he,iw,ji,ps,ur,yi.

You can set to "en" for testing purposes, or leave it empty (locales.rtl=) to avoid any locale be displayed from RTL.

Please review and comment !!

File: kernel/kernel-util/src/main/java/org/sakaiproject/util/ResourceLoader.java [hunk 1]
```diff
@@ -257,7 +257,16 @@ public String getLocaleDisplayName(Locale loc) {
 
 		return displayName.toString();
 	}
-   
+
```
Comment (buckett) [inline kernel/kernel-util/src/main/java/org/sakaiproject/util/ResourceLoader.java]: A bit of JavaDoc?

File: kernel/kernel-util/src/main/java/org/sakaiproject/util/ResourceLoader.java [hunk 2]
```diff
@@ -257,7 +257,16 @@ public String getLocaleDisplayName(Locale loc) {
...
+	public String getOrientation(Locale loc) {
+		String [] locales = ServerConfigurationService.getStrings("locales.rtl");
+		if (locales==null) {
+			locales = new String[]{"ar","dv","fa","ha","he","iw","ji","ps","ur","yi"};
+		}
+		HashSet<String> rtl = new HashSet<String>(Arrays.asList(locales));
```
Comment (buckett) [inline kernel/kernel-util/src/main/java/org/sakaiproject/util/ResourceLoader.java]: ```org.sakaiproject.util.ArrayUtil#contains(Object[], Object)``` is a better way of doing this as putting the array into a HashSet means you are iterating on it anyway.

With such a small number of items the performance of scanning the array should be very good anyway.

**Comment by ottenhoff** [comment]
Are we good to merge now?

**Comment by juanjmerono** [comment]
Yes I think so.

**Pull request merged**